### PR TITLE
add std.contract.Sequence

### PR DESCRIPTION
**Pull request by Radvendii**
This new function isn't so useful for applying directly, but there;s currently no way to store multiple contracts in a variable to be applied later. In other words
```nickel
'foo | [| 'foo |]
```
is to
```nickel
let Foo = [| 'foo |] in
'foo | Foo
```
as
```nickel
"foo" | std.contract.TagOrString | [| 'foo |]
```
is to
???

Well, before there were no options. with this PR we can do
```nickel
let Foo = std.contract.Sequence [ std.contract.TagOrString, [| 'foo |] ] in
"foo" | Foo
```

Eventually, it would be nice if the LSP could recognize this and get the same niceties as if you'd directly applied the two contracts.

### CODE REVIEW

**Comment by Radvendii** [comment]
related to #461

File: core/stdlib/std.ncl [hunk 1]
```diff
@@ -934,6 +934,27 @@
...
+      | doc m%"
+          Apply multiple contracts from left to right.
+
+          Type: `Array Contract -> Contract`
+          (for technical reasons, this function isn't actually statically typed)
+
+          # Examples
+
+          ```nickel
+          x | std.contract.Sequence [ Foo Bar ]
```
Comment (vkleen) [inline core/stdlib/std.ncl:947]:
```suggestion
          x | std.contract.Sequence [ Foo, Bar ]
```

**Review by vkleen — APPROVED** [review]
I like it, but I may be biased 😅

File: core/stdlib/std.ncl [hunk 2]
```diff
@@ -934,6 +934,27 @@
         "%
       = fun pred label value => if pred value then value else %blame% label,
 
+    Sequence
+      | doc m%"
+          Apply multiple contracts from left to right.
```
Comment (yannham) [inline core/stdlib/std.ncl:939]: Might be worth slightly more documentation, such as why do we need that if it's equivalent to just apply several contracts with `x | Foo | Bar`. A simple example with `Array (std.contract.sequence [Foo, Bar])` that can't be expressed currently with simple `|`s could be sufficient.

File: core/stdlib/std.ncl [hunk 3]
```diff
@@ -934,6 +934,27 @@
...
+          ```
+
+          is equivalent to
+
+          ```nickel
+          (x | Foo) | Bar
+          ```
+        "%
+      = fun contracts label value =>
+        std.array.fold_right (fun contract acc => std.contract.apply contract label acc) value contracts,
```
Comment (yannham) [inline core/stdlib/std.ncl:957]:
```suggestion
        std.array.fold_right (fun contract => std.contract.apply contract label) value contracts,
```

**Review by yannham — APPROVED** [review]

**Pull request merged**